h2o-algos/src/main/java/hex/rulefit/RuleEnsemble.java

```diff
@@ -0,0 +1,89 @@
+package hex.rulefit;
+
+import water.*;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.NewChunk;
+import water.fvec.Vec;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RuleEnsemble extends Iced {
+    
+   Rule[] rules;
+    
+    public RuleEnsemble(Rule[] rules) {
+        this.rules = rules;
+    }
+    
+    public Frame createGLMTrainFrame(Frame frame, int depth, int ntrees) {
+        Frame glmTrainFrame = new Frame(Key.make());
+        // filter rules and create a column for each tree
+        for (int i = 0; i < depth; i++) {
+            for (int j = 0; j < ntrees; j++) {
+                // filter rules according to varname
+                // varname is of structue "M" + modelId + "T" + node.getSubgraphNumber() + "N" + node.getNodeNumber()
+                String regex = "M" + i + "T" + j + "N" + "\\d+";
+                List<Rule> filteredRules = Arrays.asList(rules)
+                        .stream()
+                        .filter(rule ->  rule.varName.matches(regex))
+                        .collect(Collectors.toList());
+                
+                RuleEnsemble ruleEnsemble = new RuleEnsemble(filteredRules.toArray(new Rule[] {}));
+                Frame frameToMakeCategorical = ruleEnsemble.transform(frame);
+                Decoder mrtask = new Decoder(frameToMakeCategorical.names());
+                Vec column = mrtask.doAll(1, Vec.T_STR, frameToMakeCategorical).outputFrame().vec(0).toCategoricalVec();
+                glmTrainFrame.add("M" + i + "T" + j, column);
+            }
+        }
+        return glmTrainFrame;
+    }
+
+    public Frame transform(Frame frame) {
+        Frame transformedFrame = new Frame();
+        Frame actFrame;
+        for (Rule rule : rules) {
```

Comment:
It might be a good idea to do all the transformations in parallel instead of sequentially (eventually, not in this PR).